app/dao/notifications_dao.py

```diff
@@ -319,13 +319,13 @@ def delete_notifications_older_than_retention_by_type(notification_type, qry_lim
         days=7
     )
     services_with_data_retention = [x.service_id for x in flexible_data_retention]
-    service_ids_to_purge = db.session.query(Service.id).filter(Service.id.notin_(services_with_data_retention)).all()
+    service_ids_to_purge = Service.query.filter(Service.id.notin_(services_with_data_retention)).all()
```

Comment:
Hmm newer version of SQLAlchemy does not allow to query one field uniquely, instead of loading the whole object? Just wondering as the data payload will be high on the network, albeit not affecting performance assuming. Mostly curious on this change.

Reply:
The main point of this change is to get back a list of services rather than service ids (which I just realized means the variable name should be renamed 😅 ) Otherwise the call to ` insert_update_notification_history()` below was causing an error `psycopg2.ProgrammingError: can't adapt type 'Row'`

Reply:
It seems only IDs are used though in the service objects, right? Are there other service fields used somewhere else?

Reply:
Jimmy to look at this.